phys/module_sf_noahdrv.F

```diff
@@ -3721,7 +3743,12 @@ SUBROUTINE lsm_mosaic(DZ8W,QV3D,P8W3D,T3D,TSK,                      &
       
                 !  UTYPE_URB = UTYPE_URB2D(I,J) !urban type (low, high or industrial)
                 !  this need to be changed in the mosaic danli
-                IF (use_wudapt_lcz == 1) THEN
+                IF (use_distributed_aerodynamics) THEN
+                  IF (IVGTYP(I, J) == ISURBAN) THEN
+                    UTYPE_URB = 2
+                    FRC_URB2D(I, J) = MAX(0.1, MIN(0.9, 1 - shdavg(I, J) / 100.))
```

Comment:
should this already be done above in the initialization and real part?

Reply:
@cenlinhe To be honest I also don't know why I must do this. But as you can see, the code just below this line for wudapt also do the same thing for `FRC_URB2D` so I just follow the way the existing code was written.

```fortran
                  IF(UTYPE_URB==1) FRC_URB2D(I,J)=1.
                  IF(UTYPE_URB==2) FRC_URB2D(I,J)=0.99
                  IF(UTYPE_URB==3) FRC_URB2D(I,J)=1.00
                  IF(UTYPE_URB==4) FRC_URB2D(I,J)=0.65
                  IF(UTYPE_URB==5) FRC_URB2D(I,J)=0.7
                  IF(UTYPE_URB==6) FRC_URB2D(I,J)=0.65
                  IF(UTYPE_URB==7) FRC_URB2D(I,J)=0.3
                  IF(UTYPE_URB==8) FRC_URB2D(I,J)=0.85
                  IF(UTYPE_URB==9) FRC_URB2D(I,J)=0.3
                  IF(UTYPE_URB==10) FRC_URB2D(I,J)=0.55
                  IF(UTYPE_URB==11) FRC_URB2D(I,J)=1.
```

Reply:
this part is the issue in mosaic where the 2d spatially urban fraction is not used. So after the first timestep, the urban fraction is different compared to the initial 2d data.

Reply:
I will need to take a more detailed look at this. These FRC_URB2D assignments for each LCZ are supposed to be reading from the URBPARM_LCZ.TBL table values instead of hard-coded.

Reply:
@epn09 @tslin2 This original part of re-assigning FRC_URB2D is wrong. FRC_URB2D is initialized in module_physics_init.F and should only be read in as a intent(IN) variable without modifying. This has been done in both Noah and NoahMP driver. So this part should be removed from lsm_mosaic module.
@tslin2 you may want to test if removing this would solve your mosaic error.

Reply:
@cenlinhe 
Yes, I have tested already; it has the 2d spatial FRC_URB2D from the initial file

Reply:
@cenlinhe @tslin2 So what's the conclusion here? Should I remove the `FRC_URB2D` calculation here, or should it be treated in a separate PR?

Reply:
maybe @DanLi-BU can also comment on this.

Thanks

Reply:
I haven't read the WRF code this closely for such a long time but it's good to be back in the coding world. I don't think I added this line of `FRC_URB2D(I, J) = MAX(0.1, MIN(0.9, 1 - shdavg(I, J) / 100.))` when I implemented the mosaic and I agree FRC_URB2D should not be changed (unless there is a strong justification).

Reply:
FRC_URB2D(I, J) = MAX(0.1, MIN(0.9, 1 - shdavg(I, J) / 100.))

is proposed by @epn09 for new option to consider global distributed urban parameteres

Reply:
@DanLi-BU you can also review if you like

Reply:
Khanh has to buy me a lot of coffee for me to review the entire code :) I'll try my best to comment here and there but I can't really promise.